### PR TITLE
Log patterns

### PR DESCRIPTION
**Pull request by eugenesk24**
### Description
We are proposing to develop an Observability feature to build a framework that surfaces auto-generated patterns in the log data to facilitate detecting outlier events and speed up troubleshooting. The first iteration of this would be implemented as an ingest time processor that would generate an additional field in the index which consists of just the punctuation signature from an event stripped of all the alphanumeric data. 

Main Issue: #850 

- #852
- #853
- #854
- #888
- #855
- Look into changing breadcrumb trail to include query name
- #856
- #857
- Remove edit pencil icon from events table flyout
- Add back button to flyout
- Look into adding new tab when clicking see in events button
- Have flyout be all the way to the top
- Back on mouseover for flyout back button
- "Edit name" on hover over pencil icon
- Change rename flyout to just have form row
- Change name to Edit name in the flyout title
- Change table column to sentence case
- Change event details flyout header to match the patterns tab design
- Use tooltip for "Back" and "Edit name"
- Change table width to maximize pattern column
- Add search bar above patterns table
- Change punctuation signature to generated signature
- #858
- #859

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/log_patterns/patterns_tab.tsx [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+    "source = opensearch_dashboards_sample_data_logs | where match(request,'filebeat')";
+
+  const emptyPattern = {
+    firstTimestamp: 0,
+    lastTimestamp: 0,
+    puncSignature: '',
+    patternName: '',
+    ratio: '',
+    count: '',
+  };
```
Comment (ps48) [inline dashboards-observability/public/components/event_analytics/explorer/log_patterns/patterns_tab.tsx:205]: Let's add a type to the pattern vairable. Type can be added in constants/explorer.ts

File: dashboards-observability/public/components/event_analytics/explorer/events_views/doc_flyout.tsx [hunk 2]
```diff
@@ -69,7 +74,15 @@ export const DocFlyout = ({
 
   const flyoutHeader = (
     <EuiFlyoutHeader hasBorder>
-      <EuiFlexGroup>
+      <EuiFlexGroup alignItems="center" gutterSize="xs">
+        {backButtonExists && (
```
Comment (ps48) [inline dashboards-observability/public/components/event_analytics/explorer/events_views/doc_flyout.tsx:78]: Does `backButtonExists` indicate that log patterns is shown in the flyout? If so let's change the variable name.

File: dashboards-observability/public/components/event_analytics/explorer/events_views/docViewRow.tsx [hunk 3]
```diff
@@ -195,6 +195,8 @@ export const DocViewRow = forwardRef((props: IDocViewRowProps, ref) => {
         openTraces={openTraces}
         rawQuery={rawQuery}
         toggleSize={flyoutToggleSize}
+        backButtonExists={false}
+        onBackButtonClick={() => {}}
```
Comment (davidcui1225) [inline dashboards-observability/public/components/event_analytics/explorer/events_views/docViewRow.tsx:199]: just curious, why is `onBackButtonClick` empty? Is it a required part of DocFlyout that we have no functional use for?

File: dashboards-observability/public/components/event_analytics/explorer/log_patterns/pattern_detail_flyout.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    enableAllColumns: true,
+  };
+
+  const eventsTableColumns = [
+    {
+      field: 'timestamp',
+      name: 'Timestamp',
+      sortable: true,
+      width: '15%',
+      render: (value: any) => (
```
Comment (davidcui1225) [inline dashboards-observability/public/components/event_analytics/explorer/log_patterns/pattern_detail_flyout.tsx:141]: np: it looks like `value` here is an input to be cast as a `Date` type object, can we be more specific with its type than `any`?

File: dashboards-observability/public/components/event_analytics/explorer/log_patterns/patterns_tab.tsx [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+  puncSignature: string;
+  patternName: string;
+  ratio: string;
+  count: string;
+}
+
+export function PatternsTab(props: PatternsTabProps) {
+  const { http } = props;
+
+  // Uncomment below to enable EuiComboBox
```
Comment (davidcui1225) [inline dashboards-observability/public/components/event_analytics/explorer/log_patterns/patterns_tab.tsx:38]: Can we uncomment or delete this code block?

File: dashboards-observability/public/components/event_analytics/explorer/log_patterns/patterns_tab.tsx [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+                  setDummyTableData(fullData);
+                }
+              }}
+            >
+              Change table data
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiSpacer size="xs" />
+        {/* <Filters page="patterns" filters={filters} setFilters={setFilters} appConfigs={[]} /> */}
```
Comment (davidcui1225) [inline dashboards-observability/public/components/event_analytics/explorer/log_patterns/patterns_tab.tsx:278]: Same here- uncomment or delete if possible, or add a statement saying why its needed

**Comment by anirudha** [comment]
@eugenesk24 can we resolve the condflicts? is this PR still active
closing older test PR with conflict.

**Pull request closed without merging**